### PR TITLE
feat(tests): add VG-Lite render test

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

1. Add VG-Lite test and reference pictures.
2. Fixed BMP decoder flush D-Cache crash.
3. Determine whether the target color format is supported in `draw_evaluate` of VG-Lite.
4. Fixed the issue where the lottie component did not refresh the cache after updating the animation screen

TODO:
- [x] Investigate why canvas and lottie tests fail.
- [x] Investigate the reason for the inconsistency between the images produced by the second test and the first test.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by liamHowatt** [comment]
Nice!

Heads up, looks like it's executing the tests in the "Build" task too. I think you meant to just have them run in the "Run tests" task.

How come there are image mismatch failures already? Are the images direct clones of the ones in `ref_imgs`?

**Comment by kisvegabor** [comment]
Great idea! 
It seems like from this point it wouldn't bee too complicated to add a ThorVG based renderer either, as a replacement of  the SW renderer. What do you think?

**Comment by FASTSHIFT** [comment]
> Nice!
> 
> Heads up, looks like it's executing the tests in the "Build" task too. I think you meant to just have them run in the "Run tests" task.
> 
> How come there are image mismatch failures already? Are the images direct clones of the ones in `ref_imgs`?

I used the VG-Lite simulator to regenerate the reference image. Under normal circumstances, the before and after images should be consistent, unless a software-rendered image was used as a reference for the second comparison.

**Comment by FASTSHIFT** [comment]
> Great idea! It seems like from this point it wouldn't bee too complicated to add a ThorVG based renderer either, as a replacement of the SW renderer. What do you think?

Yes, using thorvg can achieve most of the functions of the software renderer, but there are also some limitations: 
1. Extensive use of `float` may not perform as well as the software LVGL software renderer.
2. Only images in ARGB8888 color format are supported. RGB565 and INDEX8 require additional conversion.

**Comment by kisvegabor** [comment]
In 64 bit build the vg lite ref images are used. See [here](https://github.com/lvgl/lvgl/actions/runs/9266747401/job/25491630574?pr=6264#step:7:4252)

**Comment by kisvegabor** [comment]
> Yes, using thorvg can achieve most of the functions of the software renderer, but there are also some limitations:

If we target high end MPUs too, these shouldn't be an issue. Having  better feature could be more valuable.

**Comment by FASTSHIFT** [comment]
On 32-bit systems, the anti-aliasing algorithm output of ThorVG has a slight deviation. I added a tolerance value setting.
image

**Comment by FASTSHIFT** [comment]
oooooohhhhhh!!!! Finally all the tests have passed, we can start reviewing~

**Review by liamHowatt — COMMENTED** [review]
Test cases still being run in the CI build task :wink: should only be in the CI test task. The tests are running in [here](https://github.com/lvgl/lvgl/actions/runs/9283246351/job/25543061056). They should not be.

**Comment by kisvegabor** [comment]
Coool! It's not critical, but I wonder if you are planning to make Thor-VG as a draw unit?

**Comment by kisvegabor** [comment]
By using the tolerance can we use the same reference images for both VGLite and SW render?

**Comment by FASTSHIFT** [comment]
> Coool! It's not critical, but I wonder if you are planning to make Thor-VG as a draw unit?

There is no plan yet. But I noticed that ThorVG supports OpenGLES rendering backend, so if we can successfully connect to ThorVG, then OpenGLES rendering backend will be automatically supported.

**Comment by FASTSHIFT** [comment]
> By using the tolerance can we use the same reference images for both VGLite and SW render?

Not yet, because I noticed that there are differences between the rounded corners of ThorVG and the LVGL software renderer, which causes the tolerance value to be exceeded.

image

```bash
lvgl/tests/src/test_cases/widgets/test_bar.c:190:test_bar_normal:INFO: 
Screenshot compare error
  - File: ref_imgs/widgets/bar_1.png
  - Mode: full refresh
  - At x:34, y:20.
  - Expected: 0x00F1C2C2
  - Actual:   0x00FDF8F8
  - Tolerance: 0
```

There are still some rendering errors, and there is still a lot of work to do before it is completely solved. We can consider adding software rendering screenshots for comparison after the problems are solved.

**Comment by FASTSHIFT** [comment]
@kisvegabor @liamHowatt Is there anything that needs to be modified? Can we merge it? ;)

**Comment by liamHowatt** [comment]
Please see this comment: https://github.com/lvgl/lvgl/pull/6264#pullrequestreview-2086224969

> Test cases still being run in the CI build task 😉 should only be in the CI test task. The tests are running in [here](https://github.com/lvgl/lvgl/actions/runs/9283246351/job/25543061056). They should not be.

**Comment by FASTSHIFT** [comment]
> Please see this comment: [#6264 (review)](https://github.com/lvgl/lvgl/pull/6264#pullrequestreview-2086224969)
> 
> > Test cases still being run in the CI build task 😉 should only be in the CI test task. The tests are running in [here](https://github.com/lvgl/lvgl/actions/runs/9283246351/job/25543061056). They should not be.

I see what you mean, `OPTIONS_VG_LITE` has been removed from `ccpp.yml`.

**Comment by liamHowatt** [comment]
Sorry, I should've been more clear on the way to fix it. I don't think you should remove `OPTIONS_VG_LITE` from the build list. That should stay.

I pushed a commit that does what I think makes sense. Please change it if it's incorrect, or revert it / tell me to revert it.

**Comment by FASTSHIFT** [comment]
> Sorry, I should've been more clear on the way to fix it. I don't think you should remove `OPTIONS_VG_LITE` from the build list. That should stay.
> 
> I pushed a commit that does what I think makes sense. Please change it if it's incorrect, or revert it / tell me to revert it.

Thank you for your help, your changes are very professional~

File: tests/ref_imgs/widgets/canvas_2.png
Comment (liamHowatt) [inline tests/ref_imgs/widgets/canvas_2.png:1]: I think canvas_[2-8].png can be removed?

**Review by liamHowatt — COMMENTED** [review]
Thanks!

File: tests/ref_imgs/widgets/canvas_2.png
Comment (FASTSHIFT) [reply tests/ref_imgs/widgets/canvas_2.png:1]: Done

**Review by kisvegabor — COMMENTED** [review]
Looks good to me! Just keep in mind the ThorVG draw_unit topic :slightly_smiling_face:

**Comment by FASTSHIFT** [comment]
@liamHowatt @kisvegabor @XuNeo 
Any one can approve this~ o(*￣︶￣*)o

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]
Sorry, approved, merging. :slightly_smiling_face:

**Pull request merged**